zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -177,13 +370,58 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             &mut reversion_info,
         );
 
-        // TODO: Handle creation transaction
         // TODO: Handle precompiled
 
-        let is_empty_code_hash =
-            IsEqualGadget::construct(cb, phase2_code_hash.expr(), cb.empty_hash_rlc());
-        let is_zero_code_hash = IsZeroGadget::construct(cb, phase2_code_hash.expr());
-        let is_empty_code = or::expr([is_empty_code_hash.expr(), is_zero_code_hash.expr()]);
+        // Handle contract creation transaction
+        // tx_is_create = true when tx.to = none
+        cb.condition(tx_is_create.expr(), |cb| {
+            cb.account_write(
+                call_callee_address.expr(),
+                AccountFieldTag::Nonce,
+                1.expr(),
+                0.expr(),
+                None,
```

Comment:
i am not sure whether the creation tx always success? if not we need reversion_info here.